libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1738,7 +1738,30 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 		}
 		else if (member == "name")
 		{
-			solUnimplementedAssert(false, "");
+			TypePointer arg = dynamic_cast<MagicType const&>(*_memberAccess.expression().annotation().type).typeArgument();
+			ContractDefinition const& contract = dynamic_cast<ContractType const&>(*arg).contractDefinition();
+			string contractName = contract.name();
+			size_t words = (contractName.length() + 31) / 32;
+			vector<map<string, string>> wordParams(words);
+			for (size_t i = 0; i < words; ++i)
+			{
+				wordParams[i]["offset"] = to_string(i * 32);
+				wordParams[i]["wordValue"] = formatAsStringOrNumber(contractName.substr(32 * i, 32));
```

Comment:
Don't we have some kind of helper for storing literal data? This case it should just go into a `data` section in the object.

Reply:
I didn't know it exists, but let me try searching for it. I checked in `revertStringifDebug` it was done same way.

Reply:
Regardless of how we store the string, we should have a yul util function to call with a literal string that returns a newly allocated memory array containing it.

Reply:
I moved storing string to util function.
Couldn't find helper that stores it to data section. Probably we can do that as part of other PR if needed.